(Samples) Transfer in (#231)
OfficeDev/office-scripts-docs: docs/resources/samples/range-basics.md

---
title: 'Range basics in Office Scripts'
description: 'Learn basics about using the Range object in Office Scripts.'
ms.date: 02/26/2021
localization_priority: Normal
---

> # Range Basics

Reviewer: This is a good article to merge some of the existing samples into (or vice versa).

Author: Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)

`Range` is the foundational object within the Office Scripts Excel object model. [Range APIs](/javascript/api/office-scripts/excelscript/excelscript.range) allow access to both data and format available on the grid and link other key objects within Excel such as worksheets, tables, charts, etc.

A range is identified using its address such as "A1:B4" or using a named-item, which is a named key for a given set of cells. In the Excel object model, both a cell and group of cells are referred as _range_. `Range` can contain cell-level attributes such as data within a cell and also cell and cells-level attributes such as format, borders, etc.

> `Range` can also be obtained via user's selection that consists of at least one cell. As you interact with the range, it is important to keep these cell and range relationships clear.

Reviewer 2: ```suggestion
`Range` can also be obtained via user's selection that consists of at least one cell. As you interact with the range, it's important to keep these cell and range relationships clear.
```

  inputRange.setValues(DATA);
  // Call a helper function to place border around the range.
  borderAround(inputRange);
}

/**
 * A Helper function that computes the target range given the target range's starting cell and selected range. 
 */
function computeTargetRange(targetCell: ExcelScript.Range, data: string[][]): ExcelScript.Range {
  const targetRange = targetCell.getResizedRange(data.length - 1, data[0].length - 1);
  return targetRange;
}

/**
 * A Helper function that places a border around the range.
 */
function borderAround(range: ExcelScript.Range): void {
  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeLeft).setStyle(ExcelScript.BorderLineStyle.dash);
  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeRight).setStyle(ExcelScript.BorderLineStyle.dash);
  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeTop).setStyle(ExcelScript.BorderLineStyle.dash);
  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeBottom).setStyle(ExcelScript.BorderLineStyle.dash);
  return;
}

// Values used for range setup.
const DATA = [
  ['Item', 'Bread', 'Donuts', 'Cookies', 'Cakes', 'Pies'],
  ['Amount', '2', '1.5', '4', '12', '26']
]
```

> ## Methods that return some range metadata

Reviewer: We should consider pulling this list into the main Object Model article.

Author: Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)

Author: Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)

Reviewer 3: I feel the "Methods that ..." section would be hard to maintain as we add more methods. Unless you see a lot of value in it, I'd remove it.
